dht.go

```diff
@@ -925,8 +925,12 @@ func (dht *IpfsDHT) maybeAddAddrs(p peer.ID, addrs []ma.Multiaddr, ttl time.Dura
 	if p == dht.self || dht.host.Network().Connectedness(p) == network.Connected {
 		return
 	}
-	if dht.addrFilter != nil {
-		addrs = dht.addrFilter(addrs)
+	dht.peerstore.AddAddrs(p, dht.filterAddrs(addrs), ttl)
+}
+
+func (dht *IpfsDHT) filterAddrs(addrs []ma.Multiaddr) []ma.Multiaddr {
```

Comment:
Why ? The peer id is not used in the filtering algorithm, imo this would give a false hint.

Reply:
I think that's more a matter of naming that helper. Each time you're passing a peer.AddrInfo you're doing the same and this could be made more concise imo. It's a matter of taste, so feel free to disagree - it's not important for the functionality.

Reply:
I like the interface that @Jorropo proposed. Filtering seems easier to comprehend if the same type is given as output as in input 👍🏻